lib/src/sai_redis_record.cpp

```diff
@@ -22,10 +22,34 @@ std::string getTimestamp()
 
 // recording needs to be enabled explicitly
 volatile bool g_record = false;
+volatile bool g_logrotate = false;
 
 std::ofstream recording;
 std::mutex g_recordMutex;
 
+std::string recfile = "dummy.rec";
+
+void logRotate()
+{
+    SWSS_LOG_ENTER();
+
+    recording.close();
+
+    /*
+     * On log rotate we will use the same file name, we are assuming that
+     * logrotate deamon move filename to filename.1 and we will create new
+     * empty file here.
+     */
+
+    recording.open(recfile);
```

Comment:
>open [](start = 14, length = 4)

For performance concern, refer [https://stackoverflow.com/a/164770/2514803](https://stackoverflow.com/a/164770/2514803). Just FYI.